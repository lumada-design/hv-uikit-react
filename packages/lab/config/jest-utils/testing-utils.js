/* eslint-disable import/no-extraneous-dependencies */
/* eslint-env jest */

import React from "react";

import { render } from "@testing-library/react";

import { HvProvider } from "@hitachivantara/uikit-react-core";

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => <HvProvider>{children}</HvProvider>;

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
