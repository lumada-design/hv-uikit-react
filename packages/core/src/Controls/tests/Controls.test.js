/* eslint-disable react/prop-types */
/* eslint-env jest */

import React from "react";
import { render } from "testing-utils";
import HvControls from "../Controls";

export default {
  title: "Tests/Controls",
  parameters: {
    docs: {
      disable: true,
      page: null,
    },
  },
};

const data = [
  {
    firstName: "tanner",
    lastName: "linsley",
    age: 29,
    visits: 100,
    status: "In Relationship",
    progress: 50,
  },
  {
    firstName: "derek",
    lastName: "perkins",
    age: 40,
    visits: 40,
    status: "Single",
    progress: 80,
  },
  {
    firstName: "joe",
    lastName: "bergevin",
    age: 45,
    visits: 20,
    status: "Complicated",
    progress: 10,
  },
];

function App(props) {
  const columns = React.useMemo(
    () => [
      {
        Header: "Name",
        columns: [
          {
            Header: "First Name",
            accessor: "firstName",
          },
          {
            Header: "Last Name",
            accessor: "lastName",
          },
        ],
      },
      {
        Header: "Info",
        columns: [
          {
            Header: "Age",
            accessor: "age",
          },
          {
            Header: "Visits",
            accessor: "visits",
          },
          {
            Header: "Status",
            accessor: "status",
          },
          {
            Header: "Profile Progress",
            accessor: "progress",
          },
        ],
      },
    ],
    []
  );

  return (
    <HvControls columns={columns} data={data} {...props}>
      {props?.leftControl && <HvControls.LeftControls placeholder="Search" />}
      {props?.rightControl && (
        <HvControls.RightControls
          values={[
            { id: "age", label: "Age" },
            { id: "visits", label: "Visits" },
          ]}
        />
      )}
    </HvControls>
  );
}

describe("Controls", () => {
  it("should render the component without customization", () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
  it("should render the component without left controls", () => {
    const { container } = render(<App leftControl={false} />);

    expect(container).toMatchSnapshot();
  });
  it("should render the component without right controls", () => {
    const { container } = render(<App rightControls={false} />);

    expect(container).toMatchSnapshot();
  });
});
