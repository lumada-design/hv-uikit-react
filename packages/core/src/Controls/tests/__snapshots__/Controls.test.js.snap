// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should render the component without customization 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-leftControl-2"
    />
    <div
      class="makeStyles-rightControl-3"
    />
  </div>
</div>
`;

exports[`Controls should render the component without left controls 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-leftControl-2"
    />
    <div
      class="makeStyles-rightControl-3"
    />
  </div>
</div>
`;

exports[`Controls should render the component without right controls 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-leftControl-2"
    />
    <div
      class="makeStyles-rightControl-3"
    />
  </div>
</div>
`;
